src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -740,11 +771,25 @@ trait ChoosesMode {
       //      println(s"receiving response: ${theRideHailResult}")
       val newPersonData = Some(theRideHailResult.request.requestId) match {
         case choosesModeData.rideHail2TransitAccessInquiryId =>
-          choosesModeData.copy(rideHail2TransitAccessResult = Some(theRideHailResult))
+          choosesModeData.copy(
+            rideHail2TransitAccessResult = Some(theRideHailResult),
+            routingFinished = choosesModeData.rideHail2TransitEgressResult.nonEmpty
+          )
         case choosesModeData.rideHail2TransitEgressInquiryId =>
-          choosesModeData.copy(rideHail2TransitEgressResult = Some(theRideHailResult))
+          choosesModeData.copy(
+            rideHail2TransitEgressResult = Some(theRideHailResult),
+            routingFinished = choosesModeData.rideHail2TransitAccessResult.nonEmpty
+          )
         case _ =>
-          choosesModeData.copy(rideHailResult = Some(theRideHailResult))
+          val routingFinished =
+            choosesModeData.rideHail2TransitRoutingResponse.nonEmpty &&
+            choosesModeData.routingResponse.nonEmpty &&
+            choosesModeData.rideHail2TransitAccessResult.nonEmpty &&
+            choosesModeData.rideHail2TransitEgressResult.nonEmpty
+          choosesModeData.copy(
+            rideHailResult = Some(theRideHailResult),
+            routingFinished = choosesModeData.routingFinished || routingFinished
```

Comment:
I remeber introducing this `routingFinished` but I don't see it is used now. Am I missing something?

Reply:
It seems like it's still used here, although I haven't fully understood the logic

https://github.com/LBNL-UCB-STI/beam/blob/e8acab10604ccf6a7c1f258b98e7c22bddea1dc4/src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala#L851